### PR TITLE
Added multi-gpu support (model wrapper)

### PR DESCRIPTION
**Pull request by tno123**

### CODE REVIEW

**Review by andreped — APPROVED** [review]
LGTM. Lets see if this impacts the other tests

**Comment by andreped** [comment]
This fix fails for dynamic optimizers such as `Adam` as even though we pass zero gradients it appears that the optimizer state is still changed. I believe in `Adam` that the dynamic learning rate is updated for every single step regardless of gradients.

This was mentioned as a likely issue [in this thread](https://github.com/keras-team/keras/issues/17429#issuecomment-1405307750).

Might be that there is some other way to solve this with the model wrapper.
